runtime/tests/clinkerffi/downcall.c

```diff
@@ -4669,3 +4669,155 @@ addDoublesFromMultipleStructPtrs_returnStruct(stru_2_Doubles *arg1, stru_2_Doubl
 	doubleStruct.elem2 = arg1->elem2 + arg2->elem2 + arg3->elem2;
 	return doubleStruct;
 }
+
+/**
+ * Do xor operation on each boolean of a native array (dereferenced from a pointer to a native array)
+ * and assign the result to the element of a on-heap array at the same index.
+ *
+ * @param size the array size
+ * @param heapPtr a pointer to a on-heap array
+ * @param nativePtr a pointer to a native array
+ * @return the sum of integers
+ */
+void
+setBoolFromArrayPtrWithXor(int size, char *heapPtr, char *nativePtr)
+{
+	int index = 0;
+	while (index < size) {
+		heapPtr[index] = nativePtr[index] ^ false;
```

Comment:
What is the point of the `^ false`? Besides mixing types (`char` and `bool`) this seems strange.

Reply:
The intention is to validate (in java) each element in the heap array is assigned with a different `bool` value from the native array.

Reply:
For every possible value of `nativePtr[index]`, `nativePtr[index] ^ false` is the same value; why didn't you write
```c
		heapPtr[index] = nativePtr[index];
```
instead? The result would be the same, without mixing `char` and `bool` types.

Reply:
This is something already did in the Jtreg test suite in OpenJDK at https://github.com/ibmruntimes/openj9-openjdk-jdk22/blob/openj9/test/jdk/java/foreign/critical/libCritical.c#L58, which we have to make test case different from their test code.

Reply:
I don't see a problem removing `^ false` from the test code.
Alternatively, the test could be adjusted to use the function in `libCritical.c`.